website/api/zkvm/input-output.md

```diff
@@ -0,0 +1,274 @@
+# Introduction to Input/Output in the zkVM
+
+This document provides an overview of how to perform I/O operations in the zkVM,
+going over the different file descriptors and methods used to handle inputs and
+outputs, the distinction between public and private data and additional
+considerations when working with I/O in the zkVM.
+
+As briefly mentioned in [Guest Code 101][guest-code-101-io], we can perform I/O
+between the guest and host by using some methods under the [`env`
+module][env-module-functions]. These methods allow sending private data between
+them. In the zkVM, there is a distinction between public and private data.
+Public data is included in the journal and is part of the proof, while private
+data is sent between the guest and host and is not included in the journal nor
+consists in part of the proof.
+
+## Public and Private Data
+
+We draw a distinction between public and private data in the zkVM. Whereas
+public data is included in the journal and becomes part of the proof, private
+data is only accessible by the host and guest. It's important to understand this
+distinction in order to handle data correctly and ensure that sensitive
+information is not publicly disclosed.
+
+Public data comprises everything that is included in the
+[journal][journal_terminology] as a result of a zkVM application's execution.
+Since this data is the _output_ of the application, public data is often also
+referred to as _public outputs_.
+
+On the other hand, private data is everything that is not included in the proof.
+In other words, private data is only accessible by the host and guest. This kind
+of data can be both in the form of _inputs_ from the host to the guest and
+_outputs_ from the guest to the host.
+
+While private data is not included in the proof, it is still important to take
```

Comment:
What if instead of session, we describe this situationally. We could say that "the private output is accessible to party that is generating the proof".